pkgs/development/libraries/gegl/reproducibility_fix.patch

```diff
@@ -0,0 +1,26 @@
+--- a/gegl/gegl-config.c	2023-03-26 14:19:28.000000000 +0100
++++ b/gegl/gegl-config.c	2023-05-07 13:18:51.971220727 +0100
+@@ -322,13 +322,7 @@
+     mem_available = (uint64_t) sysconf (_SC_AVPHYS_PAGES) * sysconf (_SC_PAGESIZE);
+ #endif
+ 
+-    default_tile_cache_size = mem_total;
+-    if (default_tile_cache_size > mem_available)
+-    {
+-      default_tile_cache_size = mem_available;
+-    }
+-    if (default_tile_cache_size < mem_min)
+-      default_tile_cache_size = mem_min;
++    default_tile_cache_size = 1024 * 1024 * 1024;
+ 
+   g_object_class_install_property (gobject_class, PROP_TILE_CACHE_SIZE,
+                                    g_param_spec_uint64 ("tile-cache-size",
+@@ -380,7 +374,7 @@
+                                                      "Number of threads",
+                                                      "Number of concurrent evaluation threads",
+                                                      0, GEGL_MAX_THREADS,
+-                                                     _gegl_threads,
++                                                     1,
```

Comment:
I think this might still affect the runtime default in some cases, e.g. in runtime bindings like PyGObject. I would really suggest only changing the value when GIR file is being generated.

Reply:
Where is it being generated, if not here?

Reply:
It is being generated from here but I am asking for it to only be hardcoded **while** the GIR is being generated. Presumably the default value from ParamSpec can be used by something at runtime but the value in GIR is [rather new](https://gitlab.gnome.org/GNOME/gobject-introspection/-/merge_requests/354) so it will probably be fine to hardcode it for now.

Reply:
After discussing on `#gtk` channel, it is probably fine: https://matrix.to/#/!aUhETchlgthwWVQzhi:matrix.org/$16834682892339QAddp:libera.chat?via=gnome.org&via=matrix.org&via=libera.chat

Reply:
by "it" do you mean the current state?

Reply:
Yeah, for the `threads`, at least.

Reply:
Actually, I looked at the [GLib source code](https://gitlab.gnome.org/GNOME/glib/-/blob/de2ff264540b892a7eac26805010cf27fbb28a39/gobject/gobject.c#L2155) and for `construct` properties (which `threads` is) the `ParamSpec` default value is actually used. So this is still not good enough.